### PR TITLE
app-admin/kubelet: add kubernetes kubelet

### PR DESCRIPTION
**Pull request by crawford**
Fixes https://github.com/coreos/bugs/issues/418.

### CODE REVIEW

**Comment by mischief** [comment]
how big is it? :)

lgtm

**Comment by crawford** [comment]
23MB

File: app-admin/kubelet/kubelet-1.0.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (c) 2015 CoreOS, Inc.. All rights reserved.
+# Distributed under the terms of the GNU General Public License v2
+# $Header:$
+#
+
+EAPI=5
+
+EGIT_REPO_URI="git://github.com/GoogleCloudPlatform/kubernetes"
+EGIT_COMMIT="e310e619fc1ac4f3238bf5ebe9e7033bf5d47ee2" # v1.0.2
```
Comment (marineam) [inline app-admin/kubelet/kubelet-1.0.2.ebuild]: Alternate idea, use tarballs for fetching from github when we don't really need git itself for live ebuilds and such:

https://github.com/coreos/coreos-overlay/blob/16e4eadad1415a54bd0afce45369932395cf50c1/sys-apps/efunctions/efunctions-1.ebuild

The magic in vcs-snapshot makes sure things work.

**Comment by crawford** [comment]
@aaronlevy and I verified this.

File: app-admin/kubelet/kubelet-1.0.2.ebuild (hunk 1 above)
Comment (crawford) [inline app-admin/kubelet/kubelet-1.0.2.ebuild]: Good idea

**Comment by crawford** [comment]
@marineam updated.

File: app-admin/kubelet/kubelet-1.0.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE=""
+
+RDEPEND="net-misc/socat"
+
+src_unpack() {
+	mkdir ${S} || die
+	unpack ${A}
```
Comment (marineam) [inline app-admin/kubelet/kubelet-1.0.1.ebuild]: this unpack call isn't doing anything.

File: app-admin/kubelet/kubelet-1.0.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Header:$
+#
+
+EAPI=5
+
+DESCRIPTION="Kubernetes Container Manager"
+HOMEPAGE="http://kubernetes.io/"
+KEYWORDS="amd64 arm64"
+SRC_URI="https://storage.googleapis.com/kubernetes-release/release/v${PV}/bin/linux/${ARCH}/${PN}"
```
Comment (marineam) [inline app-admin/kubelet/kubelet-1.0.1.ebuild]: Needs `".../${PN} -> ${P}"` so the binary that is downloaded and mirrored doesn't conflict with future versions.

**Comment by marineam** [comment]
Why are you skipping the build and shipping a binary?

File: app-admin/kubelet/kubelet-1.0.1.ebuild (hunk 2 above)
Comment (marineam) [inline app-admin/kubelet/kubelet-1.0.1.ebuild]: also if $S is not the unpacked source you may as well set `S="${WORKDIR}"` above.

**Comment by marineam** [comment]
LGTM

**Pull request merged**